python/tchannel/tornado/hyperbahn.py

```diff
@@ -59,31 +69,39 @@ def advertise(tchannel, service, routers):
         tchannel.peers.get(router)
 
     @tornado.gen.coroutine
-    def _register():
-        response = yield tchannel.request(service='hyperbahn').send(
-            arg1='ad',  # advertise
-            arg2='',
-            arg3=json.dumps({
-                'services': [
-                    {
-                        'serviceName': service,
-                        'cost': 0,
-                    }
-                ]
-            }),
-            headers={'as': 'json'},
-        )
-
-        if response.code not in ErrorCode:
-            # re-register every ``DEFAULT_TTL`` seconds
-            tornado.ioloop.IOLoop.current().call_later(
-                delay=DEFAULT_TTL,
-                callback=_register,
+    def _register(attempt_counter=0):
+
+        try:
+            response = yield tchannel.request(service='hyperbahn').send(
+                arg1='ad',  # advertise
+                arg2='',
+                arg3=json.dumps({
+                    'services': [
+                        {
+                            'serviceName': service,
+                            'cost': 0,
+                        }
+                    ]
+                }),
+                headers={'as': 'json'},
+                attempt_times=1,
             )
-            raise tornado.gen.Return(response)
+        except (ProtocolError, TimeoutError) as e:
+            attempt_counter += 1
+            log.error('Failed to register with Hyperbahn: %s', e)
         else:
-            log.error('Failed to register with Hyperbahn: %s', response)
-            raise NotImplementedError  # TODO figure out behavior here
+            if response.code != StatusCode.ok:
+                attempt_counter += 1
+                log.error('Failed to register with Hyperbahn: %s', response)
+            else:
+                attempt_counter = 0
+
+        delay_time = min(DEFAULT_MAX_DELAY, fuzz(attempt_counter))
```

Comment:
In the http://www.awsarchitectureblog.com/2015/03/backoff.html.  EqualJitter works better than the fullJitter.